native/core/src/parquet/mod.rs

```diff
@@ -52,7 +52,6 @@ const STR_CLASS_NAME: &str = "java/lang/String";
 /// Parquet read context maintained across multiple JNI calls.
 struct Context {
     pub column_reader: ColumnReader,
-    pub arrays: Option<(Arc<FFI_ArrowArray>, Arc<FFI_ArrowSchema>)>,
```

Comment:
This refactoring simplifies the context and the logic. We don't need to keep the array and schema pointers in the producer side.

Reply:
It is also easier to reason the array/schema release logic after this refactoring.